MigrationV3V4/Node/V4V3-user-state-adapter-sample-bot/bots/stateManagementBot.js

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityHandler } = require('botbuilder');
+
+// The accessor names for the conversation data and user profile state property accessors.
+const CONVERSATION_DATA_PROPERTY = 'conversationData';
+const USER_PROFILE_PROPERTY = 'userProfile';
+
+class StateManagementBot extends ActivityHandler {
+    constructor(conversationState, userState) {
+        super();
+        // Create the state property accessors for the conversation data and user profile.
+        this.conversationData = conversationState.createProperty(CONVERSATION_DATA_PROPERTY);
+        this.userProfile = userState.createProperty(USER_PROFILE_PROPERTY);
+
+        // The state management objects for the conversation and user state.
+        this.conversationState = conversationState;
+        this.userState = userState;
+
+        this.onMessage(async (turnContext, next) => {
+            // Get the state properties from the turn context.
+            const userProfile = await this.userProfile.get(turnContext, {});
+            const conversationData = await this.conversationData.get(
+                turnContext, { promptedForUserName: false });
+
+            if (!userProfile.name) {
+                // First time around this is undefined, so we will prompt user for name.
+                if (conversationData.promptedForUserName) {
+                    // Set the name to what the user provided.
+                    userProfile.name = turnContext.activity.text;
+
+                    // Acknowledge that we got their name.
+                    await turnContext.sendActivity(`Thanks ${ userProfile.name }.`);
+
+                    // Reset the flag to allow the bot to go though the cycle again.
+                    conversationData.promptedForUserName = false;
+                } else {
+                    // Prompt the user for their name.
+                    await turnContext.sendActivity('What is your name?');
+
+                    // Set the flag to true, so we don't prompt in the next turn.
+                    conversationData.promptedForUserName = true;
+                }
+            } else {
+                // Add message details to the conversation data.
+                conversationData.timestamp = turnContext.activity.timestamp.toLocaleString();
+                conversationData.channelId = turnContext.activity.channelId;
+
+                // Display state data.
+                await turnContext.sendActivity(`${ userProfile.name } sent: ${ turnContext.activity.text }`);
+                await turnContext.sendActivity(`Message received at: ${ conversationData.timestamp }`);
+                await turnContext.sendActivity(`Message received from: ${ conversationData.channelId }`);
+
+                // await this.userProfile.delete(turnContext);
```

Comment:
superfluous comments

Reply:
agree, but these comments came with the original sample that this one is based on so I left them in. figured since this is a sample better to err on the side of overcommunicating. thoughts?

Reply:
Good point.